src/Orient/Orient.Client/Protocol/Operations/BaseOperation.cs

```diff
@@ -28,10 +28,7 @@ public IRecordSerializer Serializer
 
         protected bool EndOfStream(BinaryReader reader)
         {
-            BindingFlags flags = BindingFlags.Instance | BindingFlags.NonPublic;
-            var length = (Int32)typeof(BufferedStream).GetField("_readLen", flags).GetValue(reader.BaseStream);
-            var pos = (Int32)typeof(BufferedStream).GetField("_readPos", flags).GetValue(reader.BaseStream);
-            return length == pos;
+            return reader.PeekChar() < 0;

```

Comment:
I think reader.PeekChar() is always return -1 because stream is not seekable that why i using reflection.

from documentation: https://msdn.microsoft.com/en-us/library/system.io.binaryreader.peekchar.aspx
```
Return Value
Type: System.Int32 The next available character, or -1 if no more characters are available or the stream does not support seeking.
```